src/Generation/IconGenerator.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BladeUI\Icons\Generation;
+
+use Closure;
+use Illuminate\Filesystem\Filesystem;
+use RuntimeException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\SingleCommandApplication;
+use Symfony\Component\Finder\SplFileInfo;
+
+final class IconGenerator
+{
+    private Filesystem $filesystem;
+
+    private string $set;
+
+    private string $root;
+
+    private string $directory;
+
+    private string $npm = '';
+
+    private string $composer = '';
+
+    /** @var IconSetConfig[] */
+    private array $iconSets = [];
+
+    private ?Closure $svgNormalizationClosure = null;
+
+    private bool $useSingleIconSet = false;
+
+    private bool $safe = false;
```

Comment:
@owenvoke - I know this feature was a concern you brought up initially.
So i'm curious what your thoughts are on the default for this feature.

Currently without explicitly using `safe` the destination folder will be nuked before processing.
As you pointed out it'd be good to keep up with upstream removals, however they can be breaking changes.

Reply:
I'll document this clearly to anyone using it. Personally I'd want to keep on pair with icon sets as removing of icons usually involve copyrights like @owenvoke indicates. You don't want those lingering in your set. Sure, could be a breaking change but an icon missing is less severe than you app not working anymore. Plus Blade Icons is very explicit about missing icons by throwing exceptions.

Reply:
@driesvints - sounds good to me!

I was wondering if I should edit the readme to document some of this stuff. I held off since there's quite a lot to cover, so was wondering if it'd make sense on the blade-ui-kit docs.

Reply:
@mallardduck I'll document everything once this is merged. I could end up refactoring some more so don't want you to do needless work. Thanks